elastic4s-tcp/src/main/scala/com/sksamuel/elastic4s/update/UpdateExecutables.scala

```diff
@@ -26,7 +26,8 @@ trait UpdateExecutables {
       builder.filter(QueryBuilderFn(t.query))
       t.requestsPerSecond.foreach(builder.setRequestsPerSecond)
       t.maxRetries.foreach(builder.setMaxRetries)
-      t.refresh.foreach(builder.refresh)
+      if (t.refresh.exists(_ == RefreshPolicy.Immediate))
+        builder.refresh(true)
```

Comment:
This is problematic because if someone choose wait, then it won't really wait until it's finished.
Maybe we should throw an exception.

Reply:
OK, I just duplicated the logic that was in DeleteExecutables. Leaving this as-is will keep those in sync. Would fixing both of those be something that would be better handled in it's own issue?

Reply:
Yes, good idea.